src/Range.cpp

```diff
@@ -179,6 +179,43 @@ bool Range::encloses (const Range& other) const
   return false;
 }
 
+
+////////////////////////////////////////////////////////////////////////////////
+// This is a standard enclosure check, except that the other range
+// need not have an end time, even if this one does.
+//
+// Detect the following enclosure cases:
+//
+// this                     [--------)
+//   A                        [----)
+//   B                          [...
+//
+// this                     [...
+//   A                        [----)
+//   B                           [--------)
+//   C                                    [--------)
+//   D                         [...
+//   E                                 [...
+//
+bool Range::segmentContains (const Range& other) const
```

Comment:
Like `encloses` in that an interval `A` may enclose a zero-width interval `Z` even though `A` and `Z` do not overlap, but unlike `encloses` this method does not require the enclosed interval to have an end time.